File: arena-backend/src/entities/device.entity.ts
import {
  Column,
  CreateDateColumn,
  DeleteDateColumn,
  Entity,
  Index,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
  UpdateDateColumn,
} from "typeorm";
import { UserEntity } from "./user.entity";

@Entity({ name: "devices" })
export class DeviceEntity {
  @PrimaryColumn({ type: "text" })
  deviceId: string;

  @Index("idx_devices_user_id")
  @Column({ type: "text" })
  userId: string;

  @ManyToOne(() => UserEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "userId", referencedColumnName: "userId" })
  user: UserEntity;

  @Column({ type: "text" })
  fcmToken: string;

  @Column({ type: "varchar", length: 20 })
  platform: string;

  @Column({ type: "varchar", length: 100, nullable: true })
  deviceModel: string | null;

  @Column({ type: "varchar", length: 50, nullable: true })
  osVersion: string | null;

  @CreateDateColumn({ type: "timestamptz" })
  createdAt: Date;

  @UpdateDateColumn({ type: "timestamptz" })
  updatedAt: Date;

  @DeleteDateColumn({ type: "timestamptz", nullable: true })
  deletedAt: Date | null;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/device/DeviceEntity.java
package app.sandori.arena.api.domain.device;

import app.sandori.arena.api.global.entity.BaseTimeEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "devices", indexes = {
        @Index(name = "idx_devices_user_id", columnList = "userId")
})
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class DeviceEntity extends BaseTimeEntity {

    @Id
    private String deviceId;

    @Column(nullable = false)
    private String userId;

    @Column(nullable = false, columnDefinition = "text")
    private String fcmToken;

    @Column(nullable = false, length = 20)
    private String platform;

    @Column(length = 100)
    private String deviceModel;

    @Column(length = 50)
    private String osVersion;

    public DeviceEntity(String deviceId, String userId, String fcmToken,
                        String platform, String deviceModel, String osVersion) {
        this.deviceId = deviceId;
        this.userId = userId;
        this.fcmToken = fcmToken;
        this.platform = platform;
        this.deviceModel = deviceModel;
        this.osVersion = osVersion;
    }
}
